jsonparser.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <sstream>      // std::istringstream
+#include <map> 
+#include <fstream>
+
+class jsonparser{
+private:
+public:
+    static std::string rFVbQ (std::string st){   // alias: returnFirstValuebetweenQuotationmarks, finds the first substd::string that sits between " marks and returns it
+        if (st.find('"') == std::string::npos) throw "invalid json syntax"; // if doesn't contain throw
+        std::string tmp = st.substr(st.find('"')+1);
+        if (tmp.find('"') == std::string::npos) throw "invalid json syntax"; // if doesn't contain throw
+        return tmp.substr(0,tmp.find('"'));
+    }
+    static std::map<std::string, std::string>  istrmInp(std::stringstream& f){ // read the given istream file and convert the data to a map
+        std::map<std::string, std::string> parsed;
+        std::string t = "";
+        while (true) {
+            getline(f, t);
+            if (t.find('{') != std::string::npos) continue;
+            if (t.find('}') != std::string::npos) break;
+            try {
+                std::string p1 = jsonparser::rFVbQ(t);
+                if ( parsed.find(p1) != parsed.end()) throw "double declaration of variable name\n";
+                parsed.insert(std::pair<std::string, std::string>(p1,t.substr(t.find(": ")+2))); //insert var name and value as std::strings
+            }
+            catch(const char* e)    {
+                std::cerr << e << '\n';
+                std::exit(-1);
+            }               
+        }
+        return parsed;
+    }
+    static std::map<std::string, std::string>  fileInp(std::string fname){ // read lines from file and convert to istream
+        std::map<std::string, std::string> parsed;
+        std::ifstream f(fname);   
+        std::string t,w;
+        if (!f) throw fname+" file does not exist!" ;
+        while (!f.eof()) {
```

Comment:
Van némi veszély ebben is, mert az eof csak az első sikertelen olvasási kísérlet után lesz érzékelve.

Reply:
Bocsánat it nem igazán értem mire gondol. A while, ami elöltesztelős, nem érzékeli hogy eof mielőtt az első ciklusa le futna? A while elött nem olvas be sehol.

Reply:
Az eof egy bit értékét ellenőrzi. Ezt a bitet viszont csak az első sikertelen művelet fogja beállítani. Vagyis csak azután derül ki, hogy nincs mit olvasni a fájlból, hogy megpróbáltunk olvasni belőle valamit. Ha pl. egy getline olvasna a fájlból egy stringbe, akkor sikertelen olvasás esetén benne marad a stringben az előzőleg beletett tartalom. Itt figyelni kell arra, nehogy azt az utolsó sort duplán dolgozzuk fel.